[Translation] Process multiple segments within a single unit.

timewasted opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Xliff 2.0 supports multiple segments within a single unit, but the
loader was only processing the first segment.  The loader will now
correctly process all segments within a unit.

timewasted committed:
Process multiple segments within a single unit.

Xliff 2.0 supports multiple segments within a single unit, but the
loader was only processing the first segment.  The loader will now
correctly process all segments within a unit.

phansys commented on src/Symfony/Component/Translation/Tests/Loader/XliffFileLoaderTest.php:
```diff
@@ -228,4 +228,33 @@ public function testLoadVersion2WithNoteMeta()
+        $resource = __DIR__.'/../fixtures/resources-2.0-multi-segment-unit.xlf';
+        $catalog = $loader->load($resource, 'en', 'domain1');
+
+        $this->assertEquals('en', $catalog->getLocale());
```
Please, whenever possible, use `assertSame()` instead.

phansys reviewed (changes requested):
Status: Needs work.

timewasted committed:
Prefer assertSame() over assertEquals()

When performing tests, prefer using assertSame() over assertEquals()
where possible.

nicolas-grekas commented:
Best review [ignoring whitespace](https://github.com/symfony/symfony/pull/26247/files?w=1).
@Nyholm OK for you?

Status: needs review

Nyholm reviewed (approved):
According to [specification](http://docs.oasis-open.org/xliff/xliff-core/v2.0/os/xliff-core-v2.0-os.html#unit) you may have one or more segments in a `unit`. The implementation looks good.

timewasted committed:
Better checking for libxml errors.

Instead of checking libxml_get_errors() for an empty array, we should
check libxml_get_last_error() for false to make the intent of the
assertion more clear.

fabpot commented:
Thank you @timewasted.
